sql: add stored procedure commands to the parser and return error

rharding6373 opened the pull request:
sql: add stored procedure commands to the parser and return error

Adds `CREATE [OR REPLACE] PROCEDURE` and `CALL` to the parser and returns a hint with an issue number if they are used.

Epic: None
Informs: #101250

Release note: None


sql: add support for c language in parser and add unimplemented issue
    
CRDB doesn't currently support functions with language C. This commit
adds enough to the parser to be able to parse functions that may use
C, which use the `| AS 'obj_file', 'link_symbol'` instead of `| AS
'definition'` function format (see postgres docs:
https://www.postgresql.org/docs/15/sql-createfunction.html). We also add
an issue number for users to refer to if they try to call a function
with the C language option.
    
Epic: None
Fixes: #101250
    
Release note: None

rharding6373 committed:
sql: add stored procedure commands to the parser and return error

Adds `CREATE [OR REPLACE] PROCEDURE` and `CALL` to the parser and
returns a hint with an issue number if they are used.

Epic: None
Informs: #101250

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101412)
<!-- Reviewable:end -->

rharding6373 committed:
sql: add support for c language in parser and add unimplemented issue

CRDB doesn't currently support functions with language C. This commit
adds enough to the parser to be able to parse functions that may use
C, which use the `| AS 'obj_file', 'link_symbol'` instead of `| AS
'definition'` function format (see postgres docs:
https://www.postgresql.org/docs/15/sql-createfunction.html). We also add
an issue number for users to refer to if they try to call a function
with the C language option.

Epic: None
Fixes: #101250

Release note: None

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1, 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101412#-:-NTK4FFz03f4SASlR6nw:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @mgartner and @rharding6373)

___
*[`-- commits` line 21 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/101412#-NTK3qC872--zydqqA9D):*
nit: the referenced issue doesn't contain any mention of C. Consider updating that issue description or open a new issue and link to it.


<!-- Sent from Reviewable.io -->

rharding6373 reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/101412#-:-NTtB7FvAZtC7Y9Vu7sN:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @mgartner and @rytaft)

___
*[`-- commits` line 21 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/101412#-NTK3qC872--zydqqA9D):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

nit: the referenced issue doesn't contain any mention of C. Consider updating that issue description or open a new issue and link to it.
</blockquote></details>

Fixed. Thanks!


<!-- Sent from Reviewable.io -->
